src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/MultiAdminClientFragment.kt

```diff
@@ -0,0 +1,142 @@
+import com.google.chip.chiptool.GenericChipDeviceListener
+import com.google.chip.chiptool.R
+import com.google.chip.chiptool.util.DeviceIdUtil
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.multiAdminClusterFabricIdEd
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.multiAdminClusterDeviceIdEd
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.discriminatorEd
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.setupPinCodeEd
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.multiAdminClustCommandStatus
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.view.multiAdminClusterUpdateAddressBtn
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.view.basicCommissioningMethodBtn
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.view.enhancedCommissioningMethodBtn
+import kotlinx.android.synthetic.main.multi_admin_client_fragment.view.revokeBtn
+import kotlinx.android.synthetic.main.on_off_client_fragment.*
+import kotlinx.coroutines.*
+
+class MultiAdminClientFragment : Fragment() {
+  private val deviceController: ChipDeviceController
+    get() = ChipClient.getDeviceController(requireContext())
+
+  private val scope = CoroutineScope(Dispatchers.Main + Job())
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    return inflater.inflate(R.layout.multi_admin_client_fragment, container, false).apply {
+      deviceController.setCompletionListener(ChipControllerCallback())
+
+      multiAdminClusterUpdateAddressBtn.setOnClickListener { updateAddressClick() }
+      basicCommissioningMethodBtn.setOnClickListener { scope.launch { sendBasicCommissioningCommandClick() } }
+      enhancedCommissioningMethodBtn.setOnClickListener { scope.launch { sendEnhancedCommissioningCommandClick() } }
+      revokeBtn.setOnClickListener { scope.launch { sendRevokeCommandClick() } }
+    }
+  }
+
+  override fun onStart() {
+    super.onStart()
+    // TODO: use the fabric ID that was used to commission the device
+    val testFabricId = "5544332211"
+    val testDiscriminaotr = "3840"
+    val testSetupPinCode = 20202021L
+    multiAdminClusterFabricIdEd.setText(testFabricId)
+    multiAdminClusterDeviceIdEd.setText(DeviceIdUtil.getLastDeviceId(requireContext()).toString())
+    discriminatorEd.setText(testDiscriminaotr)
+    setupPinCodeEd.setText(testSetupPinCode.toString())
+  }
+
+  inner class ChipControllerCallback : GenericChipDeviceListener() {
+    override fun onConnectDeviceComplete() {}
+
+    override fun onCommissioningComplete(nodeId: Long, errorCode: Int) {
+      Log.d(TAG, "onCommissioningComplete for nodeId $nodeId: $errorCode")
+    }
+
+    override fun onSendMessageComplete(message: String?) {
+      multiAdminClustCommandStatus.text = requireContext().getString(R.string.echo_status_response, message)
```

Comment:
echo_status_response is probably not appropriate here, maybe we can create a new string?

Reply:
Thank you for your review. We are going to add a new string for status updating.